lib/ansible/plugins/strategy/__init__.py

```diff
@@ -541,6 +543,8 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
                                 for listening_handler_block in iterator._play.handlers:
                                     for listening_handler in listening_handler_block.block:
                                         listeners = getattr(listening_handler, 'listen', []) or []
+                                        if not listeners:
+                                            continue
                                         templar = self.get_handler_templar(listening_handler, iterator)
```

Comment:
i would just keep a templar object and change 'get_handler_templar' to get variables and set it, avoids creating N templar objects and the 'dummy' becomes the main templar object.

Reply:
With the main templar object, would we run into the issue we had recently that the templar would not 'refresh' tests/filters?

Reply:
Although if we keep using that Templar only for handler names, it shouldn't matter.

Reply:
that would probably be an issue if include_role is in the mix, but not otherwise